pvlib/singlediode_methods.py

```diff
@@ -0,0 +1,347 @@
+"""
+Calculate single diode model currents and voltages using methods from
+J.W. Bishop (Solar Cells, 1988).
+"""
+
+from functools import partial
+import numpy as np
+try:
+    from scipy.optimize import brentq
+except ImportError:
+    brentq = NotImplemented
```

Comment:
scipy not installed - this module calls brentq, but if it's not found then it raises `ImportError('This function requires scipy')` OK?

Reply:
✔️ added a comment to explain why `import brentq` is wrapped in `try: except`

* IMHO, it's always best to put all imports at the top of a file, even if you only use it conditionally, this is also recommended in [PEP8](http://pep8.org/#imports)
* since `brentq` is used 3 places, we would have to add a test to import it 3 times, which seems unnecessary to me